cmd/ethrex_l2/src/commands/stack.rs

```diff
@@ -164,6 +191,93 @@ impl Command {
+                    .ok_or_eyre("Cannot get finalized block")?;
+
+                let event_signature = keccak("BlockCommitted(bytes32)");
+
+                loop {
+                    // Wait for a block
+                    sleep(Duration::from_secs(12));
+
+                    let logs = eth_client
+                        .get_logs(
+                            current_block,
+                            current_block,
+                            contract_address,
+                            event_signature,
+                        )
+                        .await?;
+
+                    if !logs.is_empty() {
+                        // Get parent beacon block root hash from block
+                        let block = eth_client
+                            .get_block_by_number(BlockByNumber::Number(current_block.as_u64()))
+                            .await?;
+                        let parent_beacon_hash = block
+                            .header
+                            .parent_beacon_block_root
+                            .ok_or_eyre("Unknown parent beacon root")?;
+
+                        // Get block slot from parent beacon block
+                        let parent_beacon_block =
+                            beacon_client.get_block_by_hash(parent_beacon_hash).await?;
+                        let target_slot = parent_beacon_block.message.slot + 1;
+
+                        // Get versioned hashes from transactions
+                        let mut l2_blob_hashes = vec![];
+                        for log in logs {
+                            let tx = eth_client
+                                .get_transaction_by_hash(log.transaction_hash)
+                                .await?
+                                .ok_or_eyre(format!(
+                                    "Transaction {:#x} not found",
+                                    log.transaction_hash
+                                ))?;
+                            l2_blob_hashes.extend(tx.blob_versioned_hashes.ok_or_eyre(format!(
+                                "Blobs not found in transaction {:#x}",
+                                log.transaction_hash
+                            ))?);
+                        }
+
+                        // Get blobs from block's slot and only keep L2 commitment's blobs
+                        let block_blobs = beacon_client
+                            .get_blobs_by_slot(target_slot)
+                            .await?
+                            .into_iter()
+                            .filter(|blob| l2_blob_hashes.contains(&blob.versioned_hash()))
+                            .collect::<Vec<_>>();
+
+                        for blob in block_blobs.into_iter() {
+                            let blob_path =
+                                data_dir.join(format!("{}-{}.blob", target_slot, blob.index));
+                            std::fs::write(blob_path, blob.blob)?;
+                        }
```

Comment:
Another refactor came to my mind
```suggestion
for blob in beacon_client
                            .get_blobs_by_slot(target_slot)
                            .await?
                            .into_iter()
                            .filter(|blob| l2_blob_hashes.contains(&blob.versioned_hash()))
                        {
                            let blob_path =
                                data_dir.join(format!("{}-{}.blob", target_slot, blob.index));
                            std::fs::write(blob_path, blob.blob)?;
                        }
```